crypto/cmp/cmp_ctx.c

```diff
@@ -425,7 +425,7 @@ int OSSL_CMP_CTX_set1_referenceValue(OSSL_CMP_CTX *ctx,
 
 /* Set or clear the password to be used for protecting messages with PBMAC */
 int OSSL_CMP_CTX_set1_secretValue(OSSL_CMP_CTX *ctx, const unsigned char *sec,
-                                  int len)
+                                  const int len)
```

Comment:
What is the point of marking the parameter `len` as `const` in the first place? It's passed by value, so the caller does need to care whether the value is modified inside the function.

In my opinion, it's not the function signature of the implementation which needs to be changed, it's the signature of the prototype in the header and the documentation.

```
~/src/openssl/master$ git grep -n -A 1 OSSL_CMP_CTX_set1_secretValue | grep -B 1 'int len'
crypto/cmp/cmp_ctx.c:421:int OSSL_CMP_CTX_set1_secretValue(OSSL_CMP_CTX *ctx, const unsigned char *sec,
crypto/cmp/cmp_ctx.c-422-                                  int len)
--
doc/man3/OSSL_CMP_CTX_new.pod:118: int OSSL_CMP_CTX_set1_secretValue(OSSL_CMP_CTX *ctx, const unsigned char *sec,
doc/man3/OSSL_CMP_CTX_new.pod-119-                                   const int len);
--
include/openssl/cmp.h.in:327:int OSSL_CMP_CTX_set1_secretValue(OSSL_CMP_CTX *ctx, const unsigned char *sec,
include/openssl/cmp.h.in-328-                                  const int len);
```

Reply:
That's why I ping @DDvO to get its feeling, because this way of fixing it is much larger.
This could be a merge conflict remainder or a review issue.

Reply:
constifying scalar function parameters is a bit over-zealous, I agree.
I think that in certain places, this is done to indicate an intent that the input parameter will be used unchanged, i.e. will not be "adjusted" frivolously.

Reply:
In other words: it's leaking implementation details without good cause ;-)

Reply:
> In other words: it's leaking implementation details without good cause ;-)

No, it's making a promise.

Reply:
Sorry for my somewhat delayed response, I did not notice this PR and question to me before.

I cannot recall how that `const int` came in originally and see little practical use for it
(except that it may help the implementer of the function not to accidentally modify the parameter value).

As one can easily find out from the commit history, already in commit 08dfbe0798f57ac9e9793fdfcaff54cfdf6b3359 I removed the `const`,
so please do not re-introduce it here.
BTW, @FdaSilvaYY I just found that you had commented on that PR, though on a different topic: https://github.com/openssl/openssl/pull/17284#issuecomment-997202727.

Unfortunately, `cmp.h.in` still contains
```
int OSSL_CMP_CTX_set1_secretValue(OSSL_CMP_CTX *ctx, const unsigned char *sec,
                                  const int len);
```
so I presume that you aim to make the function definition in `cmp_ctx.c` consistent with that.

I'd much prefer if the `const` instead gets removed from the declaration in the header file,
yet we then face the problem that this appears (syntactically) to be an API change/break,
while actually (semantically) it has no effect on the API (i.e., the interface and behavior for callers).

Reply:
@DDvO I push a fix in this way.